test/Microsoft.Sbom.Parsers.Spdx30SbomParser.Tests/Parser/GeneratorTests.cs

```diff
@@ -0,0 +1,233 @@
+        expectedJsonContentAsString = NormalizeString(expectedJsonContentAsString);
+
+        Assert.IsFalse(generatedJsonString.Contains("null"));
+        Assert.AreEqual(generatedJsonString, expectedJsonContentAsString);
+    }
+
+    [TestMethod]
+    public void GenerateJsonDocumentTest_Relationship()
+    {
+        var relationshipInfo = new Relationship
+        {
+            SourceElementId = "source-id",
+            TargetElementId = "target-id",
+            RelationshipType = RelationshipType.DESCRIBES,
+        };
+
+        var generatorResult = generator.GenerateJsonDocument(relationshipInfo);
+        var generatedJsonString = generatorResult.Document.RootElement.GetRawText();
+        generatedJsonString = NormalizeString(generatedJsonString);
+
+        var expectedJsonContentAsString = SbomRelationshipJsonStrings.RelationshipJsonString;
+        expectedJsonContentAsString = NormalizeString(expectedJsonContentAsString);
+
+        Assert.IsFalse(generatedJsonString.Contains("null"));
+        Assert.AreEqual(generatedJsonString, expectedJsonContentAsString);
+    }
+
+    private string NormalizeString(string input)
+    {
+        return input.Replace("\r", string.Empty)
+                    .Replace("\n", string.Empty)
+                    .Replace(" ", string.Empty);
+    }
+
+    private string ConvertJsonToRegex(string input)
+    {
+        var pattern = Regex.Escape(input);
+
+        // Replace placeholders with appropriate regex patterns
+        pattern = pattern.Replace(@"\.\*", ".*");
+
+        return pattern;
+    }
+
+    private List<Checksum> GetSampleChecksums()
+    {
+        var checksumSha1 = new Checksum
+        {
+            Algorithm = AlgorithmName.SHA1,
+            ChecksumValue = "sha1Value"
+        };
+        var checksumSha256 = new Checksum
+        {
+            Algorithm = AlgorithmName.SHA256,
+            ChecksumValue = "sha256Value"
+        };
+
+        return new List<Checksum> { checksumSha1, checksumSha256 };
+    }
+
+    private IInternalMetadataProvider CreateInternalMetadataProvider()
+    {
+        ISbomConfig sbomConfig = new SbomConfig(fileSystemMock.Object)
+        {
+            ManifestInfo = Constants.Spdx30ManifestInfo,
+            Recorder = new SbomPackageDetailsRecorder()
+        };
+
+        mockConfigHandler.Setup(c => c.TryGetManifestConfig(out sbomConfig)).Returns(true);
+
+        var config = new Configuration
+        {
+            PackageName = new ConfigurationSetting<string>("the-package-name"),
+            PackageVersion = new ConfigurationSetting<string>("the-package-version"),
+            NamespaceUriUniquePart = new ConfigurationSetting<string>("some-custom-value-here"),
+            NamespaceUriBase = new ConfigurationSetting<string>("http://sbom.microsoft"),
+            PackageSupplier = new ConfigurationSetting<string>("the-package-supplier"),
+        };
+
+        var logger = new Mock<ILogger>();
```

Comment:
I have a lot of opinions on unit tests--let me capture some ideas:
1. Test fixtures--including the mocks--are good. They make each specific test shorter and focus on just what is happening. Your `[TestInitialize]` method builds placeholder fixtures, then each test sets just what it cares about
2. Creating mocks the `MockBehavior.Strong` is good. It ensures that your tests describe what methods should be called on the mock
3. Using the mock's `VerifyAll` method in a `[TestCleanup]` function provides a clean way to ensure that every method with a `Mock.Setup` call was actually called

Reply:
I don't think 1. applies to the unit tests but 2 and 3 are great points - thanks for pointing them out and I've included those changes now